sdk/storage/azure-storage-internal-avro/src/main/java/com/azure/storage/internal/avro/implementation/AvroReaderSyncFactory.java

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.internal.avro.implementation;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Factory class for {@link AvroSyncReader}.
+ */
+public class AvroReaderSyncFactory {
+
+    /**
+     * Gets a new instance of {@link AvroSyncReader} with support for offset and thresholdIndex.
+     *
+     * @param data The ByteBuffer that contains the Avro data.
+     * @param offset The position in the ByteBuffer from where to start parsing.
+     * @param thresholdIndex The minimum index of the objects to be returned.
+     * @return An AvroSyncReader.
+     */
+    public AvroSyncReader getAvroReader(ByteBuffer data, int offset, long thresholdIndex) {
```

Comment:
May want to look into changing this to be `Supplier<ByteBuffer>`, or a custom type, which can be backed by different data formats. For example, using this with an `InputStream` may be a little clunky as you'd need to call this many times to supply all the bytes or to read the entire `InputStream` in one go.

Reply:
Will revisit this in another PR.